control-plane/mbus-api/src/v0.rs

```diff
@@ -589,9 +601,37 @@ pub struct CreateReplica {
     pub thin: bool,
     /// protocol to expose the replica over
     pub share: Protocol,
+    /// Managed by our control plane
+    pub managed: bool,
+    /// Owners of the resource
+    pub owners: ReplicaOwners,
 }
 bus_impl_message_all!(CreateReplica, CreateReplica, Replica, Pool);
 
+/// Replica owners which is a volume or none and a list of nexuses
+#[derive(Serialize, Deserialize, Default, Debug, Clone, PartialEq)]
+pub struct ReplicaOwners {
+    volume: Option<v0::VolumeId>,
+    nexuses: Vec<NexusId>,
+}
+impl ReplicaOwners {
+    /// Check if this replica is owned by any nexuses or a volume
+    pub fn is_owned(&self) -> bool {
+        self.volume.is_some() || !self.nexuses.is_empty()
+    }
+    /// Check if this replica is owned by this volume
+    pub fn owned_by(&self, id: &v0::VolumeId) -> bool {
+        self.volume.as_ref() == Some(id)
+    }
+    /// Create new owners from the volume Id
+    pub fn from_volume(volume: &VolumeId) -> Self {
```

Comment:
Should this function just be called `new`?

Reply:
Done